src/lib/db/users.ts

```diff
@@ -0,0 +1,43 @@
+import { prisma } from "@/lib/prisma";
+import { Prisma } from "../../../generated/prisma";
+
+export async function createUser(data: Prisma.UserCreateInput) {
+  return prisma.user.create({
+    data,
+  });
+}
+
+export async function findUser(userId: string) {
+  return prisma.user.findUnique({
+    where: { id: userId },
+  });
+}
+
+export async function findUserByEmail(email: string) {
+  return prisma.user.findUnique({
+    where: { email },
+  });
+}
+
+export async function findOrCreateUserByEmail(email: string) {
+  let user = await findUserByEmail(email);
+
+  if (!user) {
+    user = await createUser({ email });
+  }
+
+  return user;
+}
```

Comment:
This is the only function that really matters. I could have done it with an `upsert` call but I threw the CRUD functions in here too and decided to just use them.